tensorflow-keras/src/main/java/org/tensorflow/keras/backend/K.java

```diff
@@ -0,0 +1,713 @@
+        return false;
+      }
+    }
+    return true;
+  }
+
+  // TODO these are "nn" ops
+  public static Operand softmax_cross_entropy_with_logits(Ops tf, Operand labels, Operand logits) {
+    return softmax_cross_entropy_with_logits(tf, labels, logits, -1);
+  }
+
+  public static Operand softmax_cross_entropy_with_logits(
+      Ops tf, Operand labels, Operand logits, int axis) {
+    axis = axis % logits.asOutput().shape().numDimensions();
+    if (axis < 0) {
+      axis += logits.asOutput().shape().numDimensions();
+    }
+
+    Operand minusOne = tf.constant(-1);
+    Operand precise_logits = logits;
+    Operand one = tf.constant(1L);
+
+    boolean convertToFloat32 =
+        logits.asOutput().dataType() == TFloat16.DTYPE
+            || logits.asOutput().dataType() == TBfloat16.DTYPE;
+    if (convertToFloat32) {
+      precise_logits = tf.dtypes.cast(logits, TFloat32.DTYPE);
+    }
+    DataType dtype = precise_logits.asOutput().dataType();
+    labels = tf.dtypes.cast(labels, dtype);
+    Operand inputRank = tf.dtypes.cast(tf.rank(precise_logits), TInt64.DTYPE);
+    Operand inputRankMinusOne = tf.dtypes.cast(tf.math.sub(inputRank, one), TInt64.DTYPE);
+    Shape shape = logits.asOutput().shape();
+
+    // Move the dim to the end if dim is not the last dimension.
+    if (axis != -1 && axis != precise_logits.asOutput().shape().numDimensions() - 1) {
+      precise_logits = moveDimToEnd(tf, precise_logits, axis, inputRank);
+      labels = moveDimToEnd(tf, labels, axis, inputRank);
+    }
+
+    Shape inputShape = precise_logits.asOutput().shape();
+    precise_logits = flattenOuterDims(tf, precise_logits);
+    labels = flattenOuterDims(tf, labels);
+    SoftmaxCrossEntropyWithLogits smax =
+        tf.nn.softmaxCrossEntropyWithLogits(precise_logits, labels);
+    Operand cost = smax.loss();
+    Operand outputShape =
+        tf.slice(
+            tf.constant(inputShape.asArray()),
+            tf.constant(new long[] {0}),
+            tf.constant(new long[] {inputShape.numDimensions() - 1}));
+    cost = tf.reshape(cost, outputShape);
+    if (tf.scope().env().isGraph() && !shape.hasUnknownDimension()) {
+      long[] array = shape.asArray();
+      long[] newArray = new long[array.length - 1];
+      if (axis < 0) {
+        axis = shape.numDimensions() + axis;
+      }
+      for (int i = 0; i < axis; i++) {
+        newArray[i] = shape.size(i);
+      }
+      for (int i = axis + 1; i < shape.numDimensions(); i++) {
+        newArray[i - 1] = shape.size(i);
+      }
+      Shape newShape = Shape.of(newArray);
+      cost = tf.reshape(cost, tf.constant(newShape.asArray()));
+    }
+
+    if (convertToFloat32) {
+      cost = tf.dtypes.cast(cost, logits.asOutput().dataType());
+    }
+    return cost;
+  }
+
+  public static <T extends TType> Operand<T> map(
+      Operand<T> input, Function<Operand<T>, Operand<T>> mapFunc) {
+    return null;
```

Comment:
We should be able to support this soon :)

Reply:
Will this be the equivalent of tf.map_fn() in Python?

Reply:
FYI, I did this implementation. It may not be as efficient as tf.map_fn()
```
public static <T extends TType> Operand<T> map(Ops tf, Operand<T> input,
            Function<Operand<T>, Operand<T>> mapFunc ) {
        Split split = tf.split(tf.constant(0), input, 1L);
        List<Operand<T>> result = new ArrayList<>();
        split.iterator().forEachRemaining(e -> 
                   result.add( mapFunc.apply((Operand<T>)e)));
        return (Operand<T>)tf.stack(result);
    }
```